Google.Api.Gax.Grpc/ApiMetadata.cs

```diff
@@ -54,22 +62,46 @@ private ApiMetadata(string name)
         /// </remarks>
         /// <param name="name">The name of the API. Must not be null or empty.</param>
         /// <param name="descriptors">The protobuf descriptors of the API. Must not be null.</param>
-        public ApiMetadata(string name, IEnumerable<FileDescriptor> descriptors) : this(name)
+        public ApiMetadata(string name, IEnumerable<FileDescriptor> descriptors) : this(name, BuildDescriptorsProviderFromDescriptors(descriptors), false)
         {
-            var actualDescriptors = descriptors.ToList().AsReadOnly();
-            _fileDescriptorsProvider = new Lazy<IReadOnlyList<FileDescriptor>>(() => actualDescriptors);
         }
 
+        /// <summary>
+        /// Method used in the above constructor to create the lazy provider.
+        /// (Unfortunately this can't be a local method.)
+        /// </summary>
+        private static Lazy<IReadOnlyList<FileDescriptor>> BuildDescriptorsProviderFromDescriptors(IEnumerable<FileDescriptor> descriptors)
+        {
+            var actualDescriptors = descriptors.ToList().AsReadOnly();
+            return new Lazy<IReadOnlyList<FileDescriptor>>(() => actualDescriptors);
+        }
+        
         /// <summary>
         /// Creates an API descriptor which lazily requests the protobuf descriptors when <see cref="ProtobufDescriptors"/> is first called.
         /// </summary>
         /// <param name="name">The name of the API. Must not be null or empty.</param>
         /// <param name="descriptorsProvider">A provider function for the protobuf descriptors of the API. Must not be null, and must not
         /// return a null value. This will only be called once by this API descriptor, when first requested.</param>
-        public ApiMetadata(string name, Func<IEnumerable<FileDescriptor>> descriptorsProvider) : this(name)
+        public ApiMetadata(string name, Func<IEnumerable<FileDescriptor>> descriptorsProvider) : this(name, BuildDescriptorsProviderFromOtherProvider(descriptorsProvider), false)
+        {
+        }
+
+        /// <summary>
+        /// Method used in the above constructor to create the lazy provider.
+        /// (Unfortunately this can't be a local method.)
```

Comment:
Or just write the whole instruction when calling the constructor, a comment there "Create a lazy provider" should be good enough?

Reply:
Not really sure what you mean - but we can't change the constructor parameter now, as this is already used in code.